### PR TITLE
DAOS-7100 tests: Re-enable pool/destroy_rebuild.py

### PR DESCRIPTION
**Pull request by saurabhtandan**
Re-enable pool/destroy_rebuild.py and fix some design issues

Skip-unit-tests: true
Skip-nlt: true
Test-tag: destroypoolrebuild

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]

**Review by ravalsam — APPROVED** [review]

File: src/tests/ftest/pool/destroy_rebuild.yaml [hunk 1]
```diff
@@ -14,13 +14,15 @@ hosts:
...
   name: daos_server
-  scm_size: 16428800
+  scm_size: 134217728
   control_method: dmg
+  pool_query_timeout: 30
 testparams:
-  rank_to_kill: 0
-  # rank_to_kill: 1
-  # rank_to_kill: 3
+  rank_to_kill:
```
Comment (sylviachanoiyee) [inline src/tests/ftest/pool/destroy_rebuild.yaml:25]: Remove comment here since it is skipped via cancelforticket?

Comment (sylviachanoiyee) [reply src/tests/ftest/pool/destroy_rebuild.yaml:25]: So I think we can turn this into mux:
testparams:
  ranks: !mux
    rank1:
      rank: 1
    rank2:
      rank: 2
    rank3:
      rank: 3

File: src/tests/ftest/pool/destroy_rebuild.py [hunk 2]
```diff
@@ -28,34 +29,36 @@ def test_destroy_while_rebuilding(self):
...
-        # Destroy the pool while rebuild is active
-        self.pool.destroy()
-
-        # Confirm the rebuild completes
-        self.pool.wait_for_rebuild(False)
-        self.log.info("Test Passed")
+        targets = self.params.get("targets", "/run/server_config/servers/*")
+        ranks = self.params.get("rank_to_kill", "/run/testparams/*")
+
+        for rank in ranks:
```
Comment (sylviachanoiyee) [inline src/tests/ftest/pool/destroy_rebuild.py:41]: Remove the for loop here if we are muxing rank

File: src/tests/ftest/pool/destroy_rebuild.py [hunk 3]
```diff
@@ -28,34 +29,36 @@ def test_destroy_while_rebuilding(self):
...
-        self.pool.wait_for_rebuild(True)
-
-        # Destroy the pool while rebuild is active
-        self.pool.destroy()
-
-        # Confirm the rebuild completes
-        self.pool.wait_for_rebuild(False)
-        self.log.info("Test Passed")
+        targets = self.params.get("targets", "/run/server_config/servers/*")
+        ranks = self.params.get("rank_to_kill", "/run/testparams/*")
```
Comment (sylviachanoiyee) [inline src/tests/ftest/pool/destroy_rebuild.py:39]: s/ranks/rank if we are using muxing for rank

Comment (saurabhtandan) [reply src/tests/ftest/pool/destroy_rebuild.py:39]: Keeping ranks as not changing the test to muxing.

File: src/tests/ftest/pool/destroy_rebuild.yaml (hunk 1 above)
Comment (saurabhtandan) [reply src/tests/ftest/pool/destroy_rebuild.yaml:25]: Not turning into mux as the intent of the test was to stop multiple ranks together.

**Review by shimizukko — APPROVED** [review]

**Review by ravalsam — APPROVED** [review]

**Pull request merged**